subprojects/language-groovy/src/main/java/org/gradle/api/tasks/javadoc/Groovydoc.java

```diff
@@ -344,18 +350,80 @@ public void setOverviewText(@Nullable TextResource overviewText) {
     }
 
     /**
-     * Returns whether to include all classes and members (i.e. including private ones).
+     * Returns whether to include classes with private access and above.
+     *
+     * @deprecated Equivalent to calling {@link #getAccess()}{@code .get()} and checking for equivalence with {@link GroovydocAccess#PRIVATE}
      */
-    @Input
+    @Internal
+    @Deprecated
     public boolean isIncludePrivate() {
-        return includePrivate;
+        DeprecationLogger.deprecateMethod(Groovydoc.class, "isIncludePrivate()")
+            .replaceWith("getAccess()")
+            .willBeRemovedInGradle8()
+            .withUpgradeGuideSection(7, "groovydoc_option_improvements")
+            .nagUser();
```

Comment:
I missed this on the first pass, this should use `deprecateProperty` like we've done in other tasks.

```suggestion
        DeprecationLogger.deprecateProperty(Groovydoc.class, "includePrivate")
            .replaceWith("access")
            .willBeRemovedInGradle8()
            .withUpgradeGuideSection(7, "groovydoc_option_improvements")
            .nagUser();
```